route/src/route_generator_worker.cpp

```diff
@@ -438,5 +442,65 @@ namespace route {
         return true;
     }
 
+bool RouteGeneratorWorker::crossTrackErrorCheck(const geometry_msgs::PoseStampedConstPtr& msg, lanelet::ConstLanelet current)
+{
+  if (!world_model_->getMap()) 
+  {
+    ROS_ERROR_STREAM("WMListener received a route before a map was available. Dropping route message.");
+    return true;
+  }
+
+  auto route = world_model_->getRoute()->shortestPath();
+
+  bool out_of_llt_bounds = false;
+
+    lanelet::BasicPoint2d position;
+
+    position.x()= msg->pose.position.x;
+    position.y()= msg->pose.position.y;  
+
+    if (boost::geometry::distance(position, current.polygon2d()) > cross_track_dist)
+        {
+            cte_count++;
+
+            if(cte_count > 4)
+                return true;
+        }
+
+  if(boost::geometry::within(position, current.polygon2d()) == false) //Determine whether or not the vehicle is in the lanelet polygon
+      {
+        out_of_llt_bounds = true;
+      }
+
+    auto following_llts = world_model_->getMapRoutingGraph()->following(current);
+    lanelet::ConstLanelets filtered_llts;//TODO: Filter lanelets from following lanelets that are not a part ot the route
+    int out_count = 0;
+    for(auto i:following_llts)
+    {
+       
+        if (boost::geometry::within(position, i.polygon2d()) == false) 
+           out_count++;
+
+        if (boost::geometry::distance(position, current.polygon2d()) > 2.0)
```

Comment:
let's make this a parameter

Reply:
also let's use downtrack distance. See https://github.com/usdot-fhwa-stol/carma-platform/blob/fff790f974b53de0d0b2303dd7b63dfa88facc20/route/src/route_generator_worker.cpp#L348

Reply:
And only find downtrack distance to those that are on the route. Once found acceptable, we have to check the crosstrack distance again to that lanelet, the following lanelet of current one. If it is within threshold we return true. If nothing returned true in the end, we return false.